js/modules/GLPI.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * The core GLPI module code.
+ * This code should be loaded globally for all pages.
+ */
+
+// Save any previous value of GLPI.
+if (window.GLPI !== undefined) {
+    window._GLPI = window.GLPI;
+}
+
+window.GLPI = new class GLPI {
+    constructor() {
+        /** @type {Object.<string, GLPIModule>} */
+        this.modules = {};
+
+        this.known_modules = {
+            'clipboard': 'Clipboard',
+            'dialogs': 'Dialogs',
+            'rack': 'DCIM/Rack',
+        };
+
+        /** @type {EventTarget} */
+        this.event_target = new EventTarget();
+    }
+
+    async initialize() {
+        await this.registerCoreModules();
+    }
+
+    async registerCoreModules() {
+        await Promise.all([
+            this.registerKnownModule('clipboard'),
+            this.registerKnownModule('dialogs'),
+        ]);
+    }
+
+    async registerKnownModule(key) {
+        if (this.known_modules[key] !== undefined) {
+            let m = this.known_modules[key];
+            if (typeof m === 'string') {
+                // Prevent directory traversal and other issues (Remove all characters that are not a-zA-Z0-9_- or forward slash)
+                m = m.replace(/[^a-zA-Z\d_\-/]/g, '');
+                // Load the module from a script
+                const version = Math.random();
```

Comment:
With this, it will not be possible to use an outdated cached version of the file, but it will also force fetching of the file on server, even if the cached version is the good one.

To use a `v` param that is always the same for a given version, without exposing the GLPI version itself, we could use something like:
```suggestion
                const version = GLPI_FRONTEND_VERSION;
```
combined with:
```diff
diff --git a/src/Html.php b/src/Html.php
index 3c44e7e548..fc0abd20ec 100644
--- a/src/Html.php
+++ b/src/Html.php
@@ -6502,9 +6502,12 @@ HTML;
         }
         $plugins_path = 'var GLPI_PLUGINS_PATH = ' . json_encode($plugins_path) . ';';
 
+        $glpi_version = 'var GLPI_FRONTEND_VERSION = "' . FrontEnd::getVersionCacheKey(GLPI_VERSION) . '";';
+
         return self::scriptBlock("
             $cfg_glpi
             $plugins_path
+            $glpi_version
         ");
     }
 
```